client/beefy/src/round.rs

```diff
@@ -27,6 +27,10 @@ use beefy_primitives::{
 use sp_arithmetic::traits::AtLeast32BitUnsigned;
 use sp_runtime::traits::MaybeDisplay;
 
+#[cfg(test)]
+#[path = "round_tests.rs"]
+mod tests;
```

Comment:
Perhaps better to make `round` a directory with `mod.rs` file and `tests.rs` file? I find this `path` annotations a bit confusing, but I know it's already used in 2 other places, so I'd prefer consistency everywhere over addressing this one specific grumble.

Reply:
I assume we agree, that unit test should reside in a separate file. It is just easier to work with them this way.

The issue I have with the directory based approach is that I think it doesn't scale. You would end up with a separate directory for each file / module, with in many cases just contains two files. This is at least cumbersome.

Having the tests in a `[module]_tests.rs` file is, IMO, straightforward, consistent and obvious. Agreed, the `path` attribute may be a bit unfamiliar in the beginning, but that should go away pretty quickly.

Reply:
@acatangiu any preferences regarding the organisation of unit tests? Especially w.r.t. keeping everything in one file vs. splitting module implementation and unit tests up into separate files?

Reply:
I am personally used to having the tests live in the same file as the code/module they exercise, and I also see that as being mostly the norm across substrate. Having said that, it's no hard preference on my side.

> Having the tests in a [module]_tests.rs file is, IMO, straightforward, consistent and obvious. Agreed, the path attribute may be a bit unfamiliar in the beginning, but that should go away pretty quickly.

If we do split the tests in one or multiple separate files, I agree with/prefer this ^ option.

Reply:
As this is not what we do normally, I would please ask you to change this to something that doesn't incorporate "path" attribute.